unittest/unit/update-bcd.ts

```diff
@@ -861,5 +861,205 @@ describe('BCD updater', () => {
+      });
+
+      const mirroringCase = ({support, downstreamResult}) => {
+        const reports: Report[] = [
+          {
+            __version: '0.3.1',
+            results: {
+              'https://mdn-bcd-collector.appspot.com/tests/': [
+                {
+                  name: 'api.FakeInterface',
+                  exposure: 'Window',
+                  result: downstreamResult
+                }
+              ]
+            },
+            userAgent:
+              'Mozilla/5.0 (Linux; Android 10; SM-G960U) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.5112.97 Mobile Safari/537.36'
+          }
+        ];
+        const supportMatrix = getSupportMatrix(reports, browsers, []);
+        const bcd = bcdFromSupport(support);
+        update(bcd, supportMatrix, {});
+        return bcd;
+      };
+
+      describe('supported upstream (without flags)', () => {
+        it('supported in downstream test results', () => {
+          const actual = mirroringCase({
+            support: {
+              chrome: {version_added: '85'},
+              chrome_android: 'mirror'
+            },
+            downstreamResult: true
+          });
+          assert.deepEqual(
+            actual,
+            bcdFromSupport({
+              chrome: {version_added: '85'},
+              chrome_android: {version_added: '86'}
+            })
+          );
+        });
+
+        it('unsupported in downstream test results', () => {
+          const actual = mirroringCase({
+            support: {
+              chrome: {version_added: '85'},
+              chrome_android: 'mirror'
+            },
+            downstreamResult: false
+          });
+          assert.deepEqual(
+            actual,
+            bcdFromSupport({
+              chrome: {version_added: '85'},
+              chrome_android: false
+            })
+          );
+        });
+
+        it('omitted from downstream test results', () => {
+          const actual = mirroringCase({
+            support: {
+              chrome: {version_added: '85'},
+              chrome_android: 'mirror'
+            },
+            downstreamResult: null
+          });
+          assert.deepEqual(
+            actual,
+            bcdFromSupport({
+              chrome: {version_added: '85'},
+              chrome_android: 'mirror'
+            })
+          );
+        });
+      });
+
+      describe('supported upstream (with flags)', () => {
+        it('supported in downstream test results', () => {
+          const actual = mirroringCase({
+            support: {
+              chrome: {version_added: '85', flags: [{}]},
+              chrome_android: 'mirror'
+            },
+            downstreamResult: true
+          });
+          assert.deepEqual(
+            actual,
+            bcdFromSupport({
+              chrome: {version_added: '85', flags: [{}]},
+              chrome_android: [
+                {version_added: '86'},
+                {flags: [{}], version_added: '85'}
```

Comment:
The guideline on this was revised in https://github.com/mdn/browser-compat-data/pull/16637, and we should just omit the flag data. If this is the current behavior a TODO somewhere is fine though.

Reply:
I'm happy to leave this for some future patch, but I don't know how to write the comment, yet. Should the flag data be persisted in the Chrome support? For example:

```
     chrome: {version_added: '85', flags: [{}]},
     chrome_android: {version_added: '85'}
```

Or should it be removed there, too? For example:

```
     chrome: {version_added: '85'},
     chrome_android: {version_added: '85'}
```

In the latter case, should the script persist the `"mirror"` value for Chrome Android? For example:

```
     chrome: {version_added: '85'},
     chrome_android: 'mirror'
```

Reply:
This is also a situation that's independent of mirroring in principle. Whether we are updating Chrome or Chrome Android data, if the current data is just flags, and we're adding an entry for unconditional support, then we should just strip out any flag data. So perhaps:

```
// TODO: Remove flag data when adding enabled-by-default data. See https://github.com/mdn/browser-compat-data/pull/16637.
```

Reply:
Thanks!